File: Algorithm_674_Longest Continuous Increasing Subsequence.cpp
#include <iostream>
#include <vector>
using namespace std;
int findLengthOfLCIS(vector<int>& nums)
{
	int max=1,size=nums.size(),cur=1;
	if (size==0) return 0;
	for (int i=1;i<size;i++)
	{
		if (nums[i]>nums[i-1]) cur++;
		else cur=1;
		if (cur>max) max=cur;
	}
	return max;
}
int main()
{
	vector<int> nums;
	nums.push_back(1);
	nums.push_back(3);
	nums.push_back(5);
	nums.push_back(4);
	nums.push_back(7);
	cout<<findLengthOfLCIS(nums);
	return 0;
}

File: Algorithm_674_Longest Continuous Increasing Subsequence.java
// 思路：直接从左到右遍历，注意for循环外部还需要判断一下result和curLen的关系
// 这对应于最长递增子序列正好以原数组的最后一个元素结束的情况，这种情况下
// 不会进入for循环内部的else分支

class Solution {
    public int findLengthOfLCIS(int[] nums) {
        if (nums.length == 0) {
            return 0;
        }
        int result = 1, lastNum = nums[0], curLen = 1;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] > lastNum) {
                curLen++;
            } else {
                result = Math.max(result, curLen);
                curLen = 1;
            }
            lastNum = nums[i];
        }
        result = Math.max(result, curLen);
        return result;
    }
}